### ISSUE TITLE
How to deploy this?

### ISSUE THREAD

**Issue by PYL29** [XuanhaoLiu/EEG2Video#32]
We are trying to run this program. We've read the papers and installed all the dependencies but the code is very confusing(at least for us). Can you make a little guide or tutorial or a demo to show how it works and what part does what?

**Comment by XuanhaoLiu** [comment]
Thanks for your interests! Sorry about my poor coding habits, the code is confusing. However, I can show you how to reproduce our results:

1. Downsample all 2-second videos to the targeted FPS, e.g., 3 FPS in our paper, resulting in 6 frames of each video. 
2. Deploy Tune-A-Video: https://github.com/showlab/Tune-A-Video. After you can use one Input Video to fine-tune Stable Diffusion, trying to use more than one Input Video to fine-tune, in our study, we use 80 (each class 2 videos) Input Videos to fine-tune Stable Diffusion. Also, you can download the fine-tuned weights on SEED-DV at our new paper MindCine's Github: https://github.com/KevinZhou6/MindCine. Remember to save the **negative prompts' embeddings**.
3. Use Stable Diffusion's VAE encoder to get the **visual latents** of each video: it will be like shape: frames × channels × H × W.
4. Use Stable Diffusion's Text encoder to get the **text embeddings** of each video's captions: it will be like shape: 77 × 768.
5. Train a vanilla Transformer, the input is EEG signal (7×C×100), and the output is **visual latents**: frames × channels × H × W.
6. Train a simple MLP, the input is EEG signal/features, mayebe (C×d), and the output is **text embeddings**: 77 × 768.
7. Train a binary classifiers (also an MLP), the input is EEG signal/features, mayebe (C×d), and the output is Fast / Slow, corresponding to the Fast/Slow task in the EEG-VP benchmark.
8. Inference Stage: get one EEG signal, use vanilla Transformer and simple MLP to get the predicted **visual latents** and **text embeddings**. Use binary classifiers to get the Fast/Slow information, on which the DANA module's $\beta$ is depend.
If you use DANA module, than add the same noise and diverse noise to the **visual latents** now to get **noise_added visual latents**. Than put **text embeddings**, **visual latents** or **noise_added visual latents**, and **negative prompts' embeddings** into fine-tuned Tune-A-Video model to generate videos.
9. Calculate the metrics: prepare the 200 generated videos inferred from EEG signals  of the test set and the 200 original videos of the test set. Than use _40_class_run_metrics.py_ to calculate the video's semantic-level and pixel-level metrics.

**Comment by PYL29** [comment]
Sorry for the late response.  I had a few questions. How can i downsample the videos? With ffmpeg? Do we need to code a way to get the latents and the MLP or is it in the repository?

**Comment by XuanhaoLiu** [comment]
Yes, you can use ffmpeg or opencv library to downsample it, it is not complicate. 

You can download the latents in https://huggingface.co/Tianyi1229/MindCine/tree/main/data, you only need to write an simple MLP.
